src/components/AuthDialog/AuthDialog.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  Snackbar,
+  Alert,
+} from "@mui/material";
+import { useCallback, useMemo, useState } from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import style from "./AuthDialog.module.css";
+import LoginForm from "../Forms/LoginForm/LoginForm";
+import RegisterForm from "../Forms/RegisterForm/RegisterForm";
+import { IAlert } from "../../Interfaces/Interfaces";
+import { AuthMode } from "../../Enums/Enums";
+
+interface IAuthDialogProps {
+  authMode: AuthMode | null;
```

Comment:
add `declare type Maybe<T> = null | T` to the `src/global.d.ts` and use `Maybe` everywhere instead of `type | null`
```suggestion
  authMode: Maybe<AuthMode>;
```

Reply:
Fixed